packages/nextjs/services/database/repositories/users.ts

```diff
@@ -249,3 +247,50 @@ export async function filterValidUserAddresses(addresses: string[]): Promise<str
     .where(inArray(users.userAddress, addresses));
   return rows.map(row => row.userAddress);
 }
+
+const CHALLENGE_NUMBER = 7; // Number of challenges currently possible
+const CHALLENGE_POINTS = 10; // Points per accepted challenge
+const BATCH_POINTS = 20; // Points for being in a batch
+const BUILD_POINTS = 5; // Points for _first_ build only
```

Comment:
We should discuss/experiment how to show the user this points system. How do we show that being in a batch is worth 20 points? Maybe a popup that gives this information, near the XP bar?

Reply:
Messing with a modal design, open to thoughts:

<img width="1716" height="891" alt="Screenshot 2025-08-06 at 8 57 44 PM" src="https://github.com/user-attachments/assets/0434322e-66b4-43eb-84e3-fa2618fa6f00" />

Reply:
This looks great to me!

Maybe "Completing" a batch instead of "Joining"